source/components/molecules/ImageUploader/ImageUploader.js

```diff
@@ -0,0 +1,267 @@
+import { ScreenWrapper } from 'app/components/molecules';
+import uploadFile from 'app/helpers/FileUpload';
+import { excludePropetiesWithKey } from 'app/helpers/Objects';
+
+const Wrapper = styled(ScreenWrapper)`
+  padding-left: 0;
+  padding-right: 0;
+  padding-top: 15px;
+  padding-bottom: 0;
+  background-color: #f5f5f5;
+`;
+const Container = styled.ScrollView`
+  padding-left: 16px;
+  padding-right: 16px;
+`;
+const UploadIconContainer = styled.View`
+  padding-left: 12px;
+  padding-right: 12px;
+`;
+const DefaultItem = styled.TouchableHighlight`
+  border-bottom-width: 1px;
+  border-color: #c3c3c3;
+  background-color: white;
+  margin-bottom: 6px;
+  border-radius: 8px;
+`;
+const ButtonContainer = styled.View`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  margin-top: 25px;
+  width: 100%;
+`;
+const List = styled.View`
+  margin-top: 24px;
+`;
+const ListHeading = styled(Heading)`
+  margin-left: 4px;
+  margin-bottom: 8px;
+`;
+
+const Flex = styled.View`
+  flex-direction: row;
+  align-items: center;
+  padding-top: 15px;
+  padding-bottom: 15px;
+  padding-left: 15px;
+  padding-right: 15px;
+`;
+
+const IconContainer = styled.View`
+  width: 64px;
+  border-top-left-radius: 12.5px;
+  border-bottom-left-radius: 12.5px;
+  margin-right: 14px;
+`;
+
+const IconFlex = styled.View`
+  flex: 1;
+  align-items: center;
+  justify-content: center;
+`;
+
+const Content = styled.View`
+  flex: 1;
+  padding: 16px 0px 16px 8px;
+`;
+
+const ImageIcon = styled.Image`
+  width: 64px;
+  height: 64px;
+`;
+
+const ImageStatus = {
+  loading: 'LOADING',
+  finished: 'FINISHED',
+  uploadError: 'UPLOAD_ERROR',
+};
+
+const ImageItem = ({ imageData, fileName, onRemove, status }) => (
+  <DefaultItem>
+    <Flex>
+      <IconContainer highlighted>
+        <IconFlex>
+          <ImageIcon source={{ uri: `data:image/jpeg;base64,${imageData}` }} />
+        </IconFlex>
+      </IconContainer>
+      <Content>
+        <Text small>{fileName}</Text>
+      </Content>
+      <UploadIconContainer>
+        {status === ImageStatus.finished ? (
+          <Icon name="cloud-upload" color="green" />
+        ) : status === ImageStatus.loading ? (
+          <ActivityIndicator size="large" color="slategray" />
+        ) : (
+          <Icon name="error" color="red" />
+        )}
+      </UploadIconContainer>
+      <TouchableHighlight onPress={onRemove}>
+        <Icon name="delete" color="#00213F" />
+      </TouchableHighlight>
+    </Flex>
+  </DefaultItem>
+);
+ImageItem.propTypes = {
+  imageData: PropTypes.any,
+  fileName: PropTypes.string,
+  onRemove: PropTypes.func,
+  status: PropTypes.oneOf(Object.values(ImageStatus)),
+};
+
+const ImageUploader = ({ heading, buttonText, images: imgs, onChange, maxImages }) => {
+  const [images, setImages] = useState([]);
+  const [imageData, setImageData] = useState([]);
+  const [loadedStatus, setLoadedStatus] = useState([]);
```

Comment:
Originally these were all in the same array, but then I broke it out so that I could update some things separately more easily. The point is also that the info that we should save to the case data is the  images array, not the other things. And yeah, a reducer with some specific actions does make sense for keeping track of the state. If you want to refactor things into a custom hook, then sure, go ahead.